sdk/core/azure-core-amqp/src/samples/java/com/azure/core/amqp/models/AmqpAnnotatedMessageJavaDocCodeSamples.java

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.amqp.models;
+
+/**
+ * Class contains sample code snippets that will be used in javadocs.
+ */
+public class AmqpAnnotatedMessageJavaDocCodeSamples {
+    /**
+     * Get message body from {@link AmqpAnnotatedMessage}.
+     */
+    public void checkBodyType() {
+        AmqpAnnotatedMessage amqpAnnotatedMessage = null;
+        // BEGIN: com.azure.core.amqp.models.AmqpBodyType.checkBodyType
+        AmqpMessageBodyType bodyType = amqpAnnotatedMessage.getBody().getBodyType();
+        AmqpMessageBody messageBody = null;
+        switch (bodyType) {
+            case DATA:
+                messageBody = AmqpMessageBody.fromData(amqpAnnotatedMessage.getBody().getFirstData());
```

Comment:
I'm curious, what are the repercussions of a user doing this: 

```messageBody = amqpAnnotatedMessage.getBody();```

Reply:
Put another way...this code is demonstrating how to switch off the body type, but to me it raises questions about what it is actually demonstrating, and what the real use case is.

Reply:
It would be good to clarify to the user in this sample code and / or Java why the code I put above is wrong, because I would guess that most people would do that by default.

Reply:
updated sample to clarify this  why user should not call this directly.